### PR TITLE
rosdep: add polyclipping and python-{genshi, tinydb, xmltodict} deps

### PR DESCRIPTION
**Pull request by CodeFinder2**
I would like to add the following package dependencies:
- polyclipping: https://packages.ubuntu.com/xenial/libpolyclipping-dev
- python-genshi: https://packages.ubuntu.com/xenial/python-genshi
- python-xmltodict: https://packages.ubuntu.com/xenial/python-xmltodict
- python-tinydb: https://pypi.python.org/pypi/tinydb

I am using these packages for working on paths polygon as well as XML processing in Python.

Please note that I've followed the guidelines [here](https://github.com/ros/rosdistro/blob/master/CONTRIBUTING.md) and [here](http://docs.ros.org/independent/api/rosdep/html/contributing_rules.html), and tested my changes accordingly.

### CODE REVIEW

File: rosdep/python.yaml [hunk 1]
```diff
@@ -3057,6 +3061,13 @@ python-tilestache:
     heisenbug: [python-tilestache]
     schrödinger’s: [python-tilestache]
   ubuntu: [tilestache]
+python-tinydb:
```
Comment (mikaelarguedas) [inline rosdep/python.yaml]: please use `-pip` suffix for packages only available via pip

File: rosdep/python.yaml [hunk 2]
```diff
@@ -3340,6 +3351,9 @@ python-xlib:
   fedora: [python-xlib]
   gentoo: [dev-python/python-xlib]
   ubuntu: [python-xlib]
+python-xmltodict:
+  ubuntu:
+    xenial: [python-xmltodict]
```
Comment (mikaelarguedas) [inline rosdep/python.yaml]: This package is available on all ubuntu platform starting with xenial, please add them all. We also usually add the debian key as well

File: rosdep/python.yaml [hunk 3]
```diff
@@ -964,6 +964,10 @@ python-gdown-pip:
   ubuntu:
     pip:
       packages: [gdown]
+python-genshi:
+  ubuntu:
+    trusty: [python-genshi]
+    xenial: [python-genshi]
```
Comment (mikaelarguedas) [inline rosdep/python.yaml]: python-genshi is available on every not EOL ubuntu, please make the entry distribution agnostic:
`ubuntu: [python-genshi]`

File: rosdep/base.yaml [hunk 4]
```diff
@@ -3463,6 +3463,10 @@ pocketsphinx:
     xenial: [pocketsphinx]
     yakkety: [pocketsphinx]
     zesty: [pocketsphinx]
+polyclipping:
```
Comment (mikaelarguedas) [inline rosdep/base.yaml]: please use `-dev` in the name to avoid confusion with the non dev version `libpolyclipping`

File: rosdep/base.yaml [hunk 5]
```diff
@@ -3463,6 +3463,10 @@ pocketsphinx:
...
+  ubuntu:
+    trusty: [libpolyclipping-dev]
+    xenial: [libpolyclipping-dev]
```
Comment (mikaelarguedas) [inline rosdep/base.yaml]: This package is available on all alive ubuntu distribution, please majke this entry distribution agnostic `ubuntu: [libpolyclipping-dev]`.
Also please add the corresponding debian entry

**Review by mikaelarguedas — CHANGES_REQUESTED** [review]

**Comment by CodeFinder2** [comment]
Thanks for your super quick review, @mikaelarguedas! Hopefully, I've addressed all of your comments. ;-)

File: rosdep/base.yaml [hunk 6]
```diff
@@ -3463,6 +3463,12 @@ pocketsphinx:
     xenial: [pocketsphinx]
     yakkety: [pocketsphinx]
     zesty: [pocketsphinx]
+polyclipping-dev:
+  debian:
+    jessie: [libpolyclipping-dev]
+    stretch: [libpolyclipping-dev]
+    sid: [libpolyclipping-dev]
```
Comment (mikaelarguedas) [inline rosdep/base.yaml]: We usually don't target rolling distros so you don't have to add sid to this list (same below)

File: rosdep/python.yaml [hunk 7]
```diff
@@ -964,6 +964,8 @@ python-gdown-pip:
   ubuntu:
     pip:
       packages: [gdown]
+python-genshi:
+  ubuntu: [python-genshi]
```
Comment (mikaelarguedas) [inline rosdep/python.yaml]: please add a debian entry here as well

**Review by mikaelarguedas — COMMENTED** [review]
Thanks for the quick iteration, I have a few more comments. The test failure was because of an alphabetical order error on sid so removing the sid entries as per review should also fix the test

**Comment by CodeFinder2** [comment]
All (new) comments addressed in my recent commit. ;-)

(Please note that there are more "sid" entries for Debian. That was the reason why I've added mine.)

**Comment by mikaelarguedas** [comment]
> All (new) comments addressed in my recent commit. ;-)

Thanks :+1: 

> (Please note that there are more "sid" entries for Debian. That was the reason why I've added mine.)

Yeah I know, I've been hesitant to [erform a cleanup and remove them all in case anybody is actually relying on the existing ones. Thanks for being thorough though!

File: rosdep/python.yaml [hunk 8]
```diff
@@ -964,6 +964,12 @@ python-gdown-pip:
   ubuntu:
     pip:
       packages: [gdown]
+python-genshi:
+  debian:
```
Comment (mikaelarguedas) [inline rosdep/python.yaml]: these are all the alive debian distros so these rules can be collapsed to: `debian: [python-genshi]` (like you did for ubuntu)

**Comment by CodeFinder2** [comment]
Your last comment should be fixed now as well. ;-) 
Thanks for your detailed comments and for being so patient.  I totally understand that it's necessary to be meticulously in this regard.

Please let me know if there are any other changes required for merging my PR.

**Review by mikaelarguedas — APPROVED** [review]

**Comment by mikaelarguedas** [comment]
I made a small fixup and it passes the tests now. Thanks @CodeFinder2 for iterating on this

**Pull request merged**